src/handleDoubleEqual.js

```diff
@@ -1,29 +0,0 @@
-import { evaluate, round } from "mathjs";
-
-const roundedResult = (expression) => {
-  const digitAfterComma = 13;
-  round(evaluate(expression), digitAfterComma);
-};
-
-const handleDoubleEqual = (result, operate, originalLastNum) => {
-  let finalResult = 0;
-  var tempResult = result;
-  // Handling double equal
-  // If I only have sign + digit(s) + dot + digit(s)
-  const regex = /-{0,1}[0123456789]*(\.[0123456789]*){0,1}/g;
-  const matches = tempResult.match(regex) || [];
-  if (matches.length > 1 && matches[0] === tempResult) {
```

Comment:
@grimxreaper these were added back later, then removed again. the question in my mind is, what gave you the confidence to remove them here, and what changed the two other times? Did you mean to have this code moved when deleting this file and it got missed?

Reply:
Yes I meant to remove it after I tested what it was actually doing, I had an incorrect assumption that I could trust this code but actually the code was never going in this if block because the condition of this if block was never met and didn't even make sense to me (the matches[0] === tempResult) part specifically. Once I took that out, it was properly returning the result again every time equals was pressed and updating lastEquation so that was part of the bug which I was trying to solve for quite a while.

Reply:
I'm wondering how you determined the condition was never met?

Reply:
Using console.logs..

Reply:
Can you go into more detail? What was in the console logs? What conditions did you try?

Reply:
I console logged what was being saved in tempResult, and in matches...
I'll try to apply that logic to these console logs, but to my knowledge that's what I did with console.log(result, button), and was suprised at it returning [Object object]

Reply:
So you used console logs and checked the test results? Did you try running it in the browser?

Reply:
Also, how would you describe what those three lines are trying to do?

Reply:
> So you used console logs and checked the test results? Did you try running it in the browser?

Yes, it gives the same result.

> Also, how would you describe what those three lines are trying to do?

The regex is being used to identify if you have a sign + digit(s) + dot + digit(s). Then we are seeing if that pattern is being found in tempResult and assigning that to `matches`. Then on the last line for the if statement, if that pattern is found, and the first character of the matches array is strictly equal to tempResult, try what is in the if block.

Reply:
> Yes, it gives the same result.

If you’re using chrome it would give a different result depending on what you pass to console log. If not, then you aren’t giving console log something that helps you see more so try giving console log something else or add another one. Let the logs become your eyes into your code's behavior. Once what you see in the logs lines up with what you see in code, you'll likely gain understanding.

> The regex is being used to identify if you have a sign + digit(s) + dot + digit(s). Then we are seeing if that pattern is being found in tempResult and assigning that to `matches`. Then on the last line for the if statement, if that pattern is found, and the first character of the matches array is strictly equal to tempResult, try what is in the if block.

That's pretty close to a technical description. If there isn't a sign will it match? If there isn't a dot will it match?

Reply:
Updated that last comment, you may need to refresh if you looked at it already.

Reply:
```
  const regex = /-{0,1}[0123456789]*(\.[0123456789]*){0,1}/g;
  const matches = tempResult.match(regex) || [];
  if (matches.length > 1 && matches[0] === tempResult) {
```
> Also I'm curious, if there isn't a sign will it match? If there isn't a dot will it match?

If by sign you mean '-', then yes even if there isn't a sign it will match, if there isn't a dot it will also match, the {0, 1} after '-' and that second group indicates that it can match 0 or 1 of what came previous to {0, 1}.

> Can you translate that into what that code trying to do? What's its goal?

Yes, the code is trying to find instances of the pattern specified  by the regex within tempResult and assign that to the variable matches, and if it doesn't find any such pattern it will assign an empty array to matches.

Then, the code's goal is to do something else within the if block if and only if at least one pattern matching the regex was found within tempResult, and that tempResult is the first character of matches.

Reply:
That description is still too technical. I want to understand the intent of the person that added the code. What were they trying to achieve. Ideas?

Reply:
We need to find the matches of that regex pattern in the equation (in this case it was tempResult) so we can then isolate the last operator, and last number in order to repeat the previous calculation when equals is pressed again.

Reply:
@grimxreaper “matches of that regex” is still too technical. That code is looking for a number. Small side note: the new array part is a common pattern so you can write `matches.length > 1` instead of `matches != null && matches.length > 1` in the if condition.

What about the `matches[0] === tempResult`. Why is that there?

Reply:
That is there so that we know that there should only be one number in the display (in order to then run the previous equation again and return the result.

Reply:
Is there a way we can change the regex so we don’t need `matches[0] === tempResult` (Okay if not sure since regex trivia is not that important)? If we could, is there a better name we could give the regex variable?